### DISCUSSION TITLE
What to do about a stale task?

### DISCUSSION THREAD

**Discussion by btakita** [ai/nanostores#231] [Q&A]
Let's say `task` was used to fetch a resource & the responses come out of order. Should nanostores have a mechanism in `task` (or another function) to handle out of order (stale) promises or should that be a user-land solution?

```js
import { task } from 'nanostores'

import { $userId, $user } from './user-stores.js'

async function fetchUser() {
  await task(async () => {
    let response = await fetch(`https://my-api/users/${$userId.get()}`)
    $user.set(await response.json())
  })
}
```

See https://github.com/nanostores/nanostores/pull/229 & https://github.com/orgs/nanostores/discussions/216

**Comment by ai** [comment]
My first thought: it should not be in the core, since strategy could be different. For `computed` it is much easier to find a one good strategy.

But we can get some optional helper for 80% of use cases.

**Comment by btakita** [comment]
The optional helper can have a callback with a `stale()` param to support the programmer to handle stale conditions. Additional params can be added to support other strategies.

**Comment by ai** [comment]
Something like:

```ts
import { task, stale } from 'nanostores'

import { $userId, $user } from './user-stores.js'

async function fetchUser() {
  await task(async () => {
    stale(isSlated => {
      let response = await fetch(`https://my-api/users/${$userId.get()}`)
      if (isSlated()) return
      $user.set(await response.json())
    })
  })
}
```

Hm, how we will track different calls?
